src/ethereum-identity-provider.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+const IdentityProvider = require('./identity-provider')
+const { Wallet, utils } = require('ethers')
+const type = 'ethereum'
+
+class EthIdentityProvider extends IdentityProvider {
+  constructor(options) {
+    super()
```

Comment:
It'd be good to introduce the member variable `this.wallet` in the constructor to make it clear that this class contains such member variable (introducing it in the createId function below obfuscates that).